spec/game_parser_spec.rb

```diff
@@ -1,19 +1,35 @@
 # frozen_string_literal: true
+
 require 'rspec'
 require_relative '../Game'
 require_relative '../player'
 require_relative '../game_parser'
-
+require 'tempfile'
+require_relative '../top_kill_report'
 
 describe GameParser do
   let(:parser) { GameParser.new }
+  let(:file) { double(:file) }
   let(:games_file) { 'games.log' }
 
+  describe 'Creates new GameParser instance' do
+    it 'creates a new GameParser instance' do
+      expect(GameParser.new).to be_an_instance_of(GameParser)
+    end
+  end
   describe '#parse' do
     it 'parses the games log and creates a hash of games' do
       expect(parser.parse(games_file)).to be_an_instance_of(Hash)
     end
 
+    it 'should remove games with no players' do
+      allow(File).to receive(:open).with(file, 'r').and_yield(file)
```

Comment:
I recommend using a fixture instead mocking it. That way your specs is requiring a much strict concern on how to open and read the file.